client/my-sites/pages/main.jsx

```diff
@@ -115,11 +116,22 @@ class PagesMain extends React.Component {
 				 * We use iso-week and iso-week-year in order to consistently change the experiment name every week.
 				 * Assumes users have a somewhat working clock but shouldn't be a problem if they don't.
 				 */ }
-				<Experiment
+				<ExperimentPreviousClient
 					name={ `explat_test_aa_weekly_calypso_${ moment
 						.utc()
 						.format( 'GGGG' ) }_week_${ moment.utc().format( 'WW' ) }` }
 				/>
+				<Experiment
+					name={ `explat_test_aa_weekly_calypso_next_client_${ moment
+						.utc()
+						.format( 'GGGG' ) }_week_${ moment.utc().format( 'WW' ) }` }
+				>
+					{ {
+						treatment: null,
+						default: null,
+						loading: null,
+					} }
```

Comment:
Some feedback about the `<Experiment>` component API: it's very unusual that `children` is an object like this. A more classic way would be three props and no children:
```jsx
<Experiment
  treatment={ <div>treating</div> }
  default={ <div>defaulting</div> }
  loading={ <div>loading</div> }
/>
```

or a render prop that directly provides the info returned from `useExperiment`:
```jsx
<Experiment>
  { ( isLoading, assignment ) => {
    if ( isLoading ) {
      return <div>loading</div>;
    }
    if ( ! assignment?.variationName ) {
      return <div>defaulting</div>;
    }
    return <div>treating</div>;
  } }
</Experiment>
```
I would personally prefer this one. It makes the `<Experiment>` implementation an one-liner and the usage is not more complex at all.

The current `<Experiment>` has the disadvantage that the `treatment` branch doesn't have access to the `experimentAssignment` info at all. Does it really never need it?

Reply:
> A more classic way would be three props and no children:

Thats a good point! I'll implement that.

> or a render prop that directly provides the info returned from useExperiment:

The idea here is that `<Experiment>` is safe and simple to use - it ensures that you have all three cases ready.

However I can see use for a `<WithExperiment>` component that does inject it in where hooks aren't available.

Reply:
Issue for this created at 593-gh-Automattic/experimentation-platform, will be fixed after the A/A test.